msm/mycroft_skills_manager.py

```diff
@@ -85,36 +85,56 @@ def __init__(self, platform='default', skills_dir=None, repo=None,
             self.sync_skills_data()
 
     def __upgrade_skills_data(self, skills_data):
-        new = {}
+        local_skills = [s for s in self.list() if s.is_local]
         if skills_data.get('version', 0) == 0:
-            new['blacklist'] = []
-            new['version'] = 1
-            new['skills'] = []
-            local_skills = [s for s in self.list() if s.is_local]
-            default_skills = [s.name for s in self.list_defaults()]
-            for skill in local_skills:
-                if 'origin' in skills_data.get(skill.name, {}):
-                    origin = skills_data[skill.name]['origin']
-                elif skill.name in default_skills:
-                    origin = 'default'
-                elif skill.url:
-                    origin = 'cli'
-                else:
-                    origin = 'non-msm'
-                beta = skills_data.get(skill.name, {}).get('beta', False)
-                entry = build_skill_entry(skill.name, origin, beta)
-                entry['installed'] = \
-                    skills_data.get(skill.name, {}).get('installed') or 0
-                if isinstance(entry['installed'], bool):
-                    entry['installed'] = 0
-
-                entry['update'] = \
-                    skills_data.get(skill.name, {}).get('updated') or 0
-
-                new['skills'].append(entry)
-            new['upgraded'] = True
+            skills_data = self.__upgrade_to_v1(skills_data, local_skills)
+        if skills_data['version'] == 1:
```

Comment:
Should this be `.get('version', 0)` like above to prevent a keyerror (since the branch above doesn't return)?

Reply:
Basically the above should have triggered and upgraded a missing version to version 1. If that hasn't happened the upgrade to version 2 wouldn't work anyway. I think a keyerror if something is completely borked is reasonable here.

Reply:
Ahh, yeah you're right; that makes sense..